quickwit-indexing/src/lib.rs

```diff
@@ -71,3 +80,126 @@ pub async fn index_data(
+            scratch_directory: scratch_directory.temp_child()?,
+            heap_size: args.heap_size,
+            commit_policy: CommitPolicy::default(), //< TODO make the commit policy configurable
+        };
+        let indexing_pipeline_params = IndexingPipelineParams {
+            index_id: index_id.clone(),
+            source_config,
+            indexer_params,
+            metastore: metastore.clone(),
+            storage_uri_resolver: storage_uri_resolver.clone(),
+        };
+        let indexing_supervisor = IndexingPipelineSupervisor::new(indexing_pipeline_params);
+        let (_supervisor_mailbox, supervisor_handler) = universe.spawn_actor(indexing_supervisor).spawn_async();
+        supervisor_handlers.push(supervisor_handler);
+        supervisor_index_ids.push(index_id);
+    }
+
+    let rest_server = start_rest_service(args.rest_socket_addr);
+    let observe_result = observe_supervisors(&supervisor_index_ids, supervisor_handlers);
+    tokio::try_join!(rest_server, observe_result)?;
+    Ok(())
+}
+
+/// Loop on supervisor handler observe and update prometheus metrics with observation.
+// TODO: remove statistics and use counters directly into actors?
+async fn observe_supervisors(index_ids: &[&String], supervisor_handlers: Vec<ActorHandle<IndexingPipelineSupervisor>>) -> anyhow::Result<()> {
+    let mut report_interval = tokio::time::interval(Duration::from_secs(1));
+    loop {
+        report_interval.tick().await;
+        let futures = supervisor_handlers
+            .iter()
+            .map(|handler| handler.observe());
+        let observations = join_all(futures).await;
+
+        for (index_id, observation) in index_ids.iter().zip(observations.iter()) {
+            if observation.obs_type != ObservationType::PostMortem {
+                COUNTERS.num_docs.with_label_values(&[index_id]).set(observation.num_docs as i64);
+                COUNTERS.num_invalid_docs.with_label_values(&[index_id]).set(observation.num_invalid_docs as i64);
+                COUNTERS.num_local_splits.with_label_values(&[index_id]).set(observation.num_local_splits as i64);
+                COUNTERS.num_published_splits.with_label_values(&[index_id]).set(observation.num_published_splits as i64);
+                COUNTERS.num_staged_splits.with_label_values(&[index_id]).set(observation.num_staged_splits as i64);
+                COUNTERS.num_uploaded_splits.with_label_values(&[index_id]).set(observation.num_uploaded_splits as i64);
+                COUNTERS.total_bytes_processed.with_label_values(&[index_id]).set(observation.total_bytes_processed as i64);
+                COUNTERS.total_size_splits.with_label_values(&[index_id]).set(observation.total_size_splits as i64);
+            }   
+        }
+        if observations.iter().all(|observation| {
+            observation.obs_type == ObservationType::PostMortem
+        }) {
+            // TODO: funny things. I got post mortem even though one actor continues to publish splits.
```

Comment:
When you see unexpected stuff, can you filll a bug, with logs etc. This is precious information!